src/components/ha-form/types.ts

```diff
@@ -25,6 +25,8 @@ export interface HaFormBaseSchema {
     suffix?: string;
     // This value will be set initially when form is loaded
     suggested_value?: HaFormData;
+    // Readonly flag is passed here from backend for config flow
+    readonly?: boolean;
```

Comment:
nitpick: isn't read only 2 words? And so it should be `read_only` and `readOnly`

Reply:
You're technically correct. `readonly` is pretty established in frontend libraries though. We won't use it directly and convert it to `disabled` so whatever core decides as long as it isn't `disabled`

Reply:
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/r/read-only gives `read-only` which seems to also be in the dictionaries.
We should probably change to `read_only` which translates to `read-only`

Reply:
I can change the var to `read_only`, but I'm confused by what you mean by "translates to `read-only`"

Reply:
I meant for end users it should say "read-only". 
I didn't look at the frontend PR really but is there a tooltip or something for the end-users to understand these are read-only on purpose?

Reply:
As far as I know there is not a tooltip.

Maybe custom description strings could be used for the options flow fields if desired, right now they are reusing the description strings from the config flow.

I thought another enhancement could be to stick these fields in an expansion panel with a description of why they are there, but reading how config/options flow is handled in history stats I couldn't yet figure out how to make that work, so punting on that for now.